v0.21: ranking rules (#974)
meilisearch/documentation: learn/core_concepts/relevancy.md

By default, ranking rules are executed in the following order:

**1. Words**
Results are sorted by **decreasing number of matched query terms** in each matching document: find documents that contain more occurrences of the query terms first.

::: note

> Be aware that the `words` rule works from right to left, so order of the query string does affect the sorting results.

Author: ```suggestion
Be aware that the `words` rule works from right to left. Therefore, the order of search results is affected by the order of the query string.
```

> For example, if someone were to search `batman dark knight`, then the `words` rule would rank documents containing all three terms first, documents containing only `batman` and `dark` second, and documents containing only `batman` third. All other documents would be ranked last.

Reviewer: > All other documents would be ranked last.

Do you mean documents that contain only the word `batman` but with typos? I prefer that we remove this sentence. Only the documents that contain `batman` will be ranked last.

> In other words, results containing just `dark` and `knight`, or `batman` and `knight`, would be sorted last, equivalent to documents that contain none of the query terms.

Reviewer: This cannot be possible, there is no way that the engine return documents that do not contain the first word as the engine tries to remove the words from the end of the query and always requires the documents to contain all of the query words (currently tested). You can remove this sentence :)

Author: Thanks for clarifying this @Kerollmops !
